src/main.cpp

```diff
@@ -4267,6 +4267,14 @@ void GridcoinServices()
         // Tally research averages.
         if ((nBestHeight % TALLY_GRANULARITY) == 0)
         {
+            // Wait for previous retired tally to finish if running.
+            // This can happen when syncing the chain.
+            if(!IsV9Enabled_Tally(nBestHeight - 1) && !bTallyFinished_retired)
```

Comment:
Do not touch the old code. Undo your edits and just add a `while`/`sleep` spin here to wait for old tally.

Reply:
You mean remove the changes to net.cpp and just busy sleep over bTallyFinished_retired here?

I think the finished bool needs to default to true still, otherwise the busy sleep might not always exit if the retired tally never ran.

Reply:
Updated PR.

Reply:
That makes sense.

Reply:
What if the blocks go real fast (initial sync) and the tally is still running when nBestHeight - 1 is already passed.
I would remove that term and leave it only on the _finished flag.